src/GafferSceneUI/LightTool.cpp

```diff
@@ -0,0 +1,1266 @@
+const float g_penumbraHandleZAxisRotation = -20.f;
+
+// Return the plug that holds the value we need to edit, and make sure it's enabled.
+
+/// \todo This currently does nothing to enable a row if is disabled. Is that worth doing?
+
+Plug *activeValuePlug( Plug *sourcePlug )
+{
+	if( auto nameValuePlug = runTimeCast<NameValuePlug>( sourcePlug ) )
+	{
+		nameValuePlug->enabledPlug()->setValue( true );
+		return nameValuePlug->valuePlug();
+	}
+	if( auto tweakPlug = runTimeCast<TweakPlug>( sourcePlug ) )
+	{
+		tweakPlug->enabledPlug()->setValue( true );
+		return tweakPlug->valuePlug();
+	}
+	return sourcePlug;
+}
+
+const char *translucentConstantFragSource()
+{
+	return
+		"#if __VERSION__ <= 120\n"
+		"#define in varying\n"
+		"#endif\n"
+		""
+		"in vec3 fragmentCs;"
+		""
+		"void main()"
+		"{"
+		"	gl_FragColor = vec4( fragmentCs, 0.375 );"
+		"}"
+	;
+}
+
+IECoreGL::MeshPrimitivePtr cube()
+{
+	static IECoreGL::MeshPrimitivePtr result;
+	if( result )
+	{
+		return result;
+	}
+
+	IECoreScene::MeshPrimitivePtr mesh = IECoreScene::MeshPrimitive::createBox(
+		Box3f( V3f( -1 ), V3f( 1 ) )
+	);
+
+	ToGLMeshConverterPtr converter = new ToGLMeshConverter( mesh );
+	result = runTimeCast<IECoreGL::MeshPrimitive>( converter->convert() );
+
+	return result;
+}
+
+IECoreScene::MeshPrimitivePtr solidAngle( float radius, float startFraction, float stopFraction, Color3f color )
+{
+	IntVectorDataPtr vertsPerPolyData = new IntVectorData;
+	IntVectorDataPtr vertIdsData = new IntVectorData;
+	V3fVectorDataPtr pData = new V3fVectorData;
+
+	std::vector<int> &vertsPerPoly = vertsPerPolyData->writable();
+	std::vector<int> &vertIds = vertIdsData->writable();
+	std::vector<V3f> &p = pData->writable();
+
+	const int numCircleDivisions = 100;
+	const int numSegments = std::max( 1, (int)ceil( abs( stopFraction - startFraction ) * numCircleDivisions ) );
+
+	p.push_back( V3f( 0, 0, radius ) );
+
+	for( int i = 0; i < numSegments + 1; ++i )
+	{
+		const float a = ( startFraction + ( stopFraction - startFraction ) * (float)i / (float)numSegments ) * 2.f * M_PI;
+		p.push_back( V3f( -sin( a ), 0, -cos( a ) ) * radius + V3f( 0, 0, radius ) );
+	}
+
+	for( int i = 0; i < numSegments; ++i )
+	{
+		vertIds.push_back( i + 1 );
+		vertIds.push_back( i + 2 );
+		vertIds.push_back( 0 );
+		vertsPerPoly.push_back( 3 );
+	}
+
+	IECoreScene::MeshPrimitivePtr solidArc = new IECoreScene::MeshPrimitive( vertsPerPolyData, vertIdsData, "linear", pData );
+	solidArc->variables["N"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Constant, new V3fData( V3f( 0, 1, 0 ) ) );
+	solidArc->variables["Cs"] = IECoreScene::PrimitiveVariable( IECoreScene::PrimitiveVariable::Constant, new Color3fData( color ) );
+
+	return solidArc;
+}
+
+class LightToolHandle : public Handle
+{
+
+	public :
+
+		/// \todo SpotLightTool (below) is not using `metadataKey` because it knows what keys it needs.
+		/// If that is going to be the case for other tools, we should remove it.
```

Comment:
Keeping the base class as pure as possible seems sensible to me, with as few assumptions about what a derived class might want to do as possible. Even `attributePattern` seems a bit specific - my original assumption was that an Inspector was the key thing the base class would require, so that we've abstracted away what is being queried.